### DISCUSSION TITLE
Show progress of running scans

### DISCUSSION THREAD

**Discussion by 0xcrypto** [projectdiscovery/nuclei#944] [Q&A] [answered]
**Is your feature request related to a problem? Please describe.**
Running nuclei on many hosts is very slow process. I know it is running but still it visually seems hanged.

**Describe the solution you'd like**
A progress bar similar to ffuf that will show how many requests sent or templates scanned so that user can have a feel of something is being done.

**Comment by ehsandeep** [comment, marked as answer]
Hi @0xcrypto, 

Nuclei have `stats` flag for the same purpose, related issues about stats  - https://github.com/projectdiscovery/nuclei/issues/664

**Reply by geeknik** [reply]
You can also use the `metrics` flag which exposes nuclei metrics on port 9092 by default.
`curl -s http://nuclei-host:9092/metrics | jq` 👍🏻 

![2021-08-14_07-00](https://user-images.githubusercontent.com/466878/129445623-f81553d7-bd98-4f2a-b9da-df65a90dbcef.png)

If you want the metrics to "auto update", you can use this command line:
`watch -t -n5 "curl -s http://nuclei-host:9092/metrics | jq`

**Reply by 0xcrypto** [reply]
aah nice. I have been abusing verbose mode till now

**Reply by nrathaus** [reply]
You can make it a bit better with some minor changes:
`watch -d -t -n5 "curl -s http://127.0.0.1:9092/metrics | jq -S"`

This will:
1. Highlight the difference - for easier "spotting" 
2. Sort the names of the keys to make it easier to track the values
